src/config.rs

```diff
@@ -3074,32 +3056,49 @@ impl TiKvConfig {
                 ));
             }
         }
+        if last_cfg.rocksdb.wal_dir != self.rocksdb.wal_dir {
+            return Err(format!(
+                "db wal dir have been changed, former db wal_dir is '{}', \
+                 current db wal_dir is '{}', please guarantee all data wal logs \
+                 have been moved to destination directory.",
+                last_cfg.rocksdb.wal_dir, self.rocksdb.wal_dir
+            ));
+        }
 
-        if last_cfg.raft_store.raftdb_path != self.raft_store.raftdb_path
-            && !last_cfg.raft_engine.enable
-        {
+        // It's possible that `last_cfg` is not fully validated.
+        let last_raftdb_dir = last_cfg
```

Comment:
config.validate should already do this adjust. 

https://github.com/tikv/tikv/blob/5ee7d037d4e2e63d710410daac81baca7e9efe69/src/config.rs#L3183-L3187

https://github.com/tikv/tikv/blob/5ee7d037d4e2e63d710410daac81baca7e9efe69/src/config.rs#L2691-L2692

Reply:
See my PR description. Validating config from v3.0 could possibly fail at this line:
https://github.com/tikv/tikv/blob/5ee7d037d4e2e63d710410daac81baca7e9efe69/src/config.rs#L3185

I didn't verify though.

Reply:
I'm not sure about the risk of config adjustment skipped by validation error, may warn this error.

Reply:
Alright.